### PR TITLE
Use Python 3.6 or later for CuPy tests

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by kmaehashi** [comment]
LGTM, but let's add a new matrix `cupy-py35-v8` to test lowest boundary Python version for v8 branch.

**Comment by asi1024** [comment]
Added `cupy-py35-v8`. PTAL.

File: run_test.py [hunk 1]
```diff
@@ -311,6 +311,30 @@ def main():
         script = './test_cupy.sh'
         use_gcc6_or_later = True
```
Comment (emcastillo) [inline run_test.py:312]: I think we need to skip this for cupy v9

**Comment by emcastillo** [comment]
Jenkins, test this please

**Review by emcastillo — APPROVED** [review]
Test errors are unrelated
